src/bin/ion/commands/generate/generator.rs

```diff
@@ -351,14 +351,27 @@ impl<'a, L: Language + 'static> CodeGenerator<'a, L> {
             .any(|it| matches!(it.constraint(), IslConstraintValue::Fields(_, _)))
         {
             self.build_structure_from_constraints(constraints, code_gen_context, isl_type)?
+        } else if constraints
+            .iter()
+            .any(|it| matches!(it.constraint(), IslConstraintValue::Element(_, _)))
+        {
```

Comment:
Can you please make sure that all of the rules for classifying the ISL types are documented in a comment in plain english?

Reply:
Yes, will ad comments for it.